src/lib/stepper/stepper-horizontal.html

```diff
@@ -8,16 +8,25 @@
          [tabIndex]="_focusIndex == i ? 0 : -1"
          (click)="step.select()"
          (keydown)="_onKeydown($event)">
-      <div class="mat-stepper-index">
+      <div class="mat-stepper-index-new" *ngIf="!step.completed || selectedIndex == i">
         {{i + 1}}
       </div>
+      <div class="mat-stepper-index-interacted" *ngIf="step.completed && selectedIndex != i">
```

Comment:
What do you mean by sharing the code (for this and `ngClass` you mentioned below)? Do you mean creating a separate header component with the template that contains these divs?